### PR TITLE
RFC: cloudwatch compute log manager

### PR DESCRIPTION
**Pull request by prha**
### Summary & Motivation
This sets up a cloudwatch compute log manager to be used in conjunction with the ecs run launcher, which instead of displaying captured logs in the dagit UI, will construct a cloudwatch URL to point to the specific run logs.

### How I Tested These Changes
Manually tested; will need to write tests before landing

### CODE REVIEW

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9429** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9429" target="_blank">Graphite</a> 
    * **PR #9956** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9956" target="_blank">Graphite</a> 
      * **PR #9957** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9957" target="_blank">Graphite</a> 
        * **PR #9958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9958" target="_blank">Graphite</a> 
          * **PR #9959** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9959" target="_blank">Graphite</a> 
            * **PR #9960** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9960" target="_blank">Graphite</a> 
              * **PR #9961** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9961" target="_blank">Graphite</a> 
                * **PR #9962** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9962" target="_blank">Graphite</a> 
                  * **PR #10045** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10045" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10045?utm_source=stack-comment).

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9429** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9429" target="_blank">Graphite</a> 
    * **PR #9956** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9956" target="_blank">Graphite</a> 
      * **PR #9957** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9957" target="_blank">Graphite</a> 
        * **PR #9958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9958" target="_blank">Graphite</a> 
          * **PR #9959** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9959" target="_blank">Graphite</a> 
            * **PR #9960** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9960" target="_blank">Graphite</a> 
              * **PR #9961** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9961" target="_blank">Graphite</a> 
                * **PR #9962** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9962" target="_blank">Graphite</a> 
                  * **PR #10045** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10045" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10045?utm_source=stack-comment).

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9429** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9429" target="_blank">Graphite</a> 
    * **PR #9956** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9956" target="_blank">Graphite</a> 
      * **PR #9957** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9957" target="_blank">Graphite</a> 
        * **PR #9958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9958" target="_blank">Graphite</a> 
          * **PR #9959** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9959" target="_blank">Graphite</a> 
            * **PR #9960** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9960" target="_blank">Graphite</a> 
              * **PR #9961** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9961" target="_blank">Graphite</a> 
                * **PR #9962** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9962" target="_blank">Graphite</a> 
                  * **PR #10045** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10045" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10045?utm_source=stack-comment).

**Review by jmsanders — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/cloudwatch_compute_log_manager.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self, log_key: List[str]
+    ) -> CapturedLogMetadata:  # pylint: disable=unused-argument
+        metadata_uri = os.getenv("ECS_CONTAINER_METADATA_URI")
+        region = os.getenv("AWS_REGION")
+        if not metadata_uri or not region:
+            return CapturedLogMetadata()
+
+        arn_id = metadata_uri.split("/")[-1].split("-")[0]
+        base_url = f"https://{region}.console.aws.amazon.com/cloudwatch/home?region={region}"
+        log_group = f"$252Fdocker-compose$252F{self._project_name}"
```
[the remark this answers is not in the document]
Comment (prha) [reply python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/cloudwatch_compute_log_manager.py:46]: yeah, this is good feedback, thanks

File: python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/cloudwatch_compute_log_manager.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+    @staticmethod
+    def from_config_value(inst_data, config_value):
+        return CloudwatchComputeLogManager(inst_data=inst_data, **config_value)
+
+    def get_contextual_log_metadata(
+        self, log_key: List[str]
+    ) -> CapturedLogMetadata:  # pylint: disable=unused-argument
+        metadata_uri = os.getenv("ECS_CONTAINER_METADATA_URI")
```
[the remark this answers is not in the document]
Comment (prha) [reply python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/cloudwatch_compute_log_manager.py:39]: This is improperly named.  This should really be the `LogsCapturedInECSThatPointToCloudwatchLogManager`.

[the remark this answers is not in the document]
Comment (prha) [reply python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/cloudwatch_compute_log_manager.py:39]: I guess I'm trying to suss out how people would want to configure their logging sidecars.

**Pull request closed without merging**